### PR TITLE
TGUI: Command Reports

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

Implements a TGUI window for the admin Create-Command-Report verb.

image
image

[You can find a demonstration video of the window here.](https://streamable.com/86xsa5)

- The window comes with a few preset command names as well as the opportunity to use a custom command name.

- The user can also set a custom announcer sound effect to accompany their command report. This uses the list of announcer keys, so the sounds are automatically set for any station trait announcer.

- Lastly, they can check a box to determine if the report is to be announced or classified.

This PR also fixes the animes announcement line not functioning properly due to bad copy+paste.

## Why It's Good For The Game

Replaces some old input windows with TGUI and introduces some new functions to command reports for (b)admins to use. 

## Changelog
:cl: Melbert
refactor: TGUI Create-Command-Report window
admin: Admins can now set an announcement sound to accompany their command reports. 
fix: The 'anime' announcement line now functions properly again. This is not related to the above changelog line whatsoever.
/:cl:

### CODE REVIEW

File: tgui/packages/tgui/interfaces/CommandReport.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            options={announcer_sounds}
+            onSelected={value => act('set_report_sound', {
+              picked_sound: value,
+            })} />
+        </Section>
+        <Section title="Set report text:" textAlign="center">
+          <TextArea
+            width="100%"
+            height="200px"
+            mb={1}
```
Comment (tralezab) [inline tgui/packages/tgui/interfaces/CommandReport.js:53]: just a tip and not a change, if you used stacks inside the section (stack and stack.item everything inside) you wouldn't have to bother with margins as much

**Review by tralezab — COMMENTED** [review]
Here's some stuff to make your tgui better (you should test) ping me on discord if anything seems off

**Comment by TheVekter** [comment]
Nuuuuuut. Please continue murdering default pop-ups.

**Comment by Mothblocks** [comment]
I think announce contents should be on its own line, as to both scale with more options being created and to let the submit report button be wide, due to its importance.

File: code/__DEFINES/sound.dm [hunk 2]
```diff
@@ -111,3 +111,23 @@
...
+	ANNOUNCER_OUTBREAK5,
+	ANNOUNCER_OUTBREAK7,
+	ANNOUNCER_POWEROFF,
+	ANNOUNCER_POWERON,
+	ANNOUNCER_RADIATION,
+	ANNOUNCER_SHUTTLECALLED,
+	ANNOUNCER_SHUTTLEDOCK,
+	ANNOUNCER_SHUTTLERECALLED,
+	ANNOUNCER_SPANOMALIES,
+	))
```
Comment (Mothblocks) [inline code/__DEFINES/sound.dm:133]:
```suggestion
))
```

File: code/modules/admin/verbs/commandreport.dm [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+			if(!command_name)
+				to_chat(ui_user, "<span class='danger'>You can't send a report with no command name.</span>")
+				return
+			if(!command_report_content)
+				to_chat(ui_user, "<span class='danger'>You can't send a report with no contents.</span>")
+				return
+			send_announcement()
+			. = TRUE
+
+	return
```
Comment (Mothblocks) [inline code/modules/admin/verbs/commandreport.dm:119]: This isn't necessary, return is implicit, though I think you should just use `return TRUE` in place of every `. = TRUE`.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: tgui/packages/tgui/interfaces/CommandReport.js [hunk 4]
```diff
@@ -60,7 +60,7 @@ export const CommandReport = (props, context) => {
                 onChange={(e, value) => act("update_report_contents", {
                   updated_contents: value,
                 })} />
-              <Stack>
+              <Stack vertical={1}>
```
Comment (stylemistake) [inline tgui/packages/tgui/interfaces/CommandReport.js:63]: This property is boolean, delete ={1}

**Pull request merged**